### ISSUE TITLE
add a LICENSE file

### ISSUE THREAD

**Issue by waldenn** [NCSU-Libraries/annona#13]
Please add a LICENSE file to the project.

Thanks for this interesting IIIF project!

**Comment by dnoneill** [comment]
@waldenn Thanks for your interest. If you do end up using the library please let me know. I am closing this issue because the license has been added.

**Comment by waldenn** [comment]
Super nice, thanks again!

Its not a priority for my own project ([conzept](https://conze.pt)), but I am in the beginning process of brainstorming a two-part system (presenter + editor) of a presentation-tool based on deep text (text fragments) / video (start/end time) / image (your project) / URL / linking with the option of TTS or text-info annotations.

Your project could be an interesting part of that I think, though my IIIF format understanding is still very limited for now. I have started to experiment a bit [here](https://conze.pt/app/present/?voice=en-GB) (like passing the voice-style from my app to annona as a first small integration step)

This "present" app could then be used like [this](https://conze.pt/explore/Zodiac?l=en&t=link&i=Q40540&u=https%3A%2F%2Fconze.pt%2Fapp%2Fpresent%2F%3Fvoice%3Den-GB&s=true)

What I think I will need for the UI is to create is a sidebar (flat or tree-structure?) with those deeplinks and a way to create that sidebar JSON data through an easy interface. Then link the presentation to a Wikidata qid and allow them to be searched and included in the encyclopedia interface. Ideally there would be some open and not too complex JSON format, to store such a presentation format. I will likely post some more thoughts on my twitter feed at some point. Your project is an inspiration for all this!

**Comment by dnoneill** [comment]
@waldenn I keep meaning to write you back. Thank you so much. This is incredibly cool and I am so glad you are able to get use out of this. Please let me know if I can be of any help.

**Comment by waldenn** [comment]
Thanks again. I've made some progress integrating IIIF images since then.

I've got a question (please feel free to skip this if its takes too much time!).

I'm trying to add a single annotation to a working Universal Viewer setup:
  [https://conze.pt/app/iiif/#?c=&m=&s=&cv=&manifest=https%3A%2F%2Fconze.pt%2Ftest%2Fmanifest.json&xywh=-1%2C-1%2C1%2C0](https://conze.pt/app/iiif/#?c=&m=&s=&cv=&manifest=https%3A%2F%2Fconze.pt%2Ftest%2Fmanifest.json)
 
I would like to add an image annotation to the above setup, by placing a rectangle on the following area (or just a zoom-in for now): "pct:58.2499,50.3486,13.1551,6.30773" (this percentage data comes from the Wikidata data property on a depict-statement)

The following two files are used in the above example:
  https://conze.pt/test/manifest.json
  https://conze.pt/test/annotations.json

Do you see what things need to be corrected in these files for the annotation to work?
(note: don't mind the "RomanCoins" ID-strings, I will replace those later with something more general.)

I looked around for a good example that I could learn from, but did not find much or they were confusing me.
The [IIIF cookbook](https://iiif.io/api/cookbook/) also does not show a clear annotation example AFAICS.

**Comment by dnoneill** [comment]
@waldenn The problem is the use of pct. The application does not read that. It only allows for xywh. I would like to integrate the use of pct into the application because it looks like it is part of the IIIF spec. Could you give me more information about what pct is? I haven't found that much about how I might be able to convert it to xywh coordinates.

**Comment by dnoneill** [comment]
@waldenn I was answering the question that I got emailed not what is in the comment. To the best of my knowledge Universal viewer does not support annotations. The only support is the search within function. This item: https://d.lib.ncsu.edu/collections/catalog/nubian-message-1995-04-01#?c=&m=&s=&cv=&xywh=-648%2C-365%2C6018%2C7287 has annotations and the best of my knowledge the search within function is the only way to use annotations.

**Comment by waldenn** [comment]
Thanks for the feedback, makes sense. Will look into your work later, to see how all this annotation magic is done.

**Comment by waldenn** [comment]
> @waldenn The problem is the use of pct. The application does not read that. It only allows for xywh. I would like to integrate the use of pct into the application because it looks like it is part of the IIIF spec. Could you give me more information about what pct is? I haven't found that much about how I might be able to convert it to xywh coordinates.

See:
- https://iiif.io/api/image/3.0/#41-region
- https://iiif.io/api/image/2.0/#region
- https://github.com/zone47/CROTOS/blob/master/lab/cropper/js/cropper.js (an example of "pct" support)
  - http://zone47.com/crotos/lab/cropper/
- https://www.wikidata.org/wiki/Property:P2677 (the wikidata property for registering the relative position within an image)

Thanks for considering adding percentage support!

**Comment by waldenn** [comment]
Also see the WikiMedia "zoomviewer" project which can create thumbnails from percentage coordinates: [source](https://github.com/toollabs/zoomviewer/tree/master/public_html)

Example code I use for zoomviewer:
 thumb = img.replace('http://commons.wikimedia.org/wiki/Special:FilePath/', '');
 thumb = 'https://tools.wmflabs.org/zoomviewer/proxy.php?iiif=' + thumb + '/' + coord + '/full/0/default.jpg';

**Comment by dnoneill** [comment]
@waldenn this work will probably get integrated soon. Annonatate (an application I have built to create annotations) will possibly soon support this and when it does I will make sure Annona supports it as well. I am waiting on someone else to do the work on pct support in Annonatate first so I do not have a timeline on that.
